source/common/network/listen_socket_impl.cc

```diff
@@ -37,6 +38,22 @@ TcpListenSocket::TcpListenSocket(const Address::InstanceConstSharedPtr& address,
   int rc = setsockopt(fd_, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on));
   RELEASE_ASSERT(rc != -1);
 
+#if defined SOL_IP && defined IP_TRANSPARENT
```

Comment:
I believe this can be moved into `if (bind_to_port) { ... }`, since as far as I recall, it's useless without `bind()`.

Reply:
One could make the same argument about the existing `setsockopt(fd_, SOL_SOCKET, SO_REUSEADDR, ...)` above. Should I move all `setsockopt` calls into `if (bind_to_port) { ... }`?